gitpod-io/gitpod-eks-guide
# Various fixes & improvements, EKS 1.22, Gitpod 2022.06.0

## Description
This PR is based on the latest commit (https://github.com/gitpod-io/gitpod-eks-guide/commit/16929769ee29f4ba0ba2494076b4ce61f2f4f8b7) in [mrz/alb-ingress](https://github.com/gitpod-io/gitpod-eks-guide/tree/mrz/alb-ingress).

It fixes all current issues we have been experiencing lately and - optionally - provides a fully guided installation with no additional manual steps required from users (dealing with load balancers, dns and ssl certs).

Includes:
 - fixes outdated helm chart issues
 - fixes nodes unable to pull images from registry-facade due to unknown CA (via bootstrap override)
 - use latest "Ubuntu 2004" AMI (need to specify a specific AMI ID due to bootstrap override)
 - fixes object storage S3 connection / permission issue (when using the bucket & iam user of registry)
 - tested with latest Gitpod release (2022.06.0)
 - auto-create load balancers (alb & nlb) with automatic dns, certificate and firewall rules - optional
 - auto-create a Let's Encrypt cert-manager ClusterIssuer - optional
 - enable SSH access on nodes using default private key in ~/.ssh - optional

## How to test
1. Configure .env - provide your target namespace, ACM certificate ARN, R53 hosted-zone ID and CREATE_LB=true.
2. Run 'make build' & 'make install' on a clean AWS account.
3. Install & run KOTS in the target namespace.
4. Provide all KOTS configuration options as directed by the guide's output (incl. uploading 'extra-config-patch.yaml')
5. Verify all pre-flight checks are green and the installation is fully working and the `proxy` service type is `NodePort`.

## Release Notes

```release-note
```

## Documentation
The EKS guide should already be linked from the Gitpod Docs page.
